src/main/java/org/jenkinsci/plugins/ssegateway/sse/EventDispatcher.java

```diff
@@ -293,12 +300,13 @@ private void scheduleRetryQueueProcessing(long delay) {
         }
         if (delay > 0) {
             try {
-                new Timer("EventDispatcher.retryProcessor").schedule( new TimerTask() {
-                    @Override
-                    public void run() {
-                        processRetries();
-                    }
-                }, delay);
+//                new Timer("EventDispatcher.retryProcessor").schedule( new TimerTask() {
+//                    @Override
+//                    public void run() {
+//                        processRetries();
+//                    }
+//                }, delay);
+                scheduledExecutorService.schedule(this::processRetries, delay, TimeUnit.MILLISECONDS);
```

Comment:
This is better, but if users really have thousands of `EventDispatcher.retryProcessor` timer threads, then it seems like too many things are going into the retry queue, the tasks in those threads are never actually completing, or nothing is leaving the retry queue. This will prevent the thread leak, but it seems like there is still a serious bug somewhere.

Maybe we could ask some of the users to temporarily increase the logging to FINE on this class to see what message are being printed in `processRetries` to understand if the retries are even working?

Also, it looks like `processRetries` tries to resend everything in `retryQueue`, and doesn't exit until the queue is empty. In that case does it even make sense to run it on more than 1 thread? I guess if you want to try to empty the queue faster? Can the dispatching even work from multiple threads or will it end up being single-threaded anyway?